### DISCUSSION TITLE
Simultaneous multiple MSA embedding

### DISCUSSION THREAD

**Discussion by hury07** [facebookresearch/esm#98] [Q&A] [answered]
How to prepare labels and MSA sequences for simultaneous multiple MSA embedding? Specifically, how to construct a dataset and a dataloader?

**Comment by tomsercu** [comment, marked as answer]
I'm not entirely sure what you mean with "simultaneous multiple MSA embedding".
For datasets and dataloaders for training, see https://pytorch.org/tutorials/beginner/basics/data_tutorial.html

**Reply by hury07** [reply]
I want to embed a batch of sequences using ESM-MSA model, then I'll have multiple MSAs as input to ESM-MSA model.

**Reply by tomsercu** [reply]
Aha got it, your question is how to construct a batch of MSAs.
Please take a look at this example notebook, cell 10 onwards, which I think does exactly what you are looking for: https://github.com/facebookresearch/esm/blob/master/examples/contact_prediction.ipynb
